src/leader.c

```diff
@@ -450,3 +450,49 @@ int leader__barrier(struct leader *l, struct barrier *barrier, barrier_cb cb)
 	}
 	return 0;
 }
+
+int leader__reset(struct leader *l, struct exec *exec, exec_cb cb)
```

Comment:
We probably want to go through a barrier before executing operations against the DB?